### ISSUE TITLE
KAGYA専用0.6B System One (Jev) を24GB単一GPU構成に統合する

### ISSUE THREAD

**Issue by upiscium** [upiscium/PROJECT-KAGYA#271]
## 背景

PROJECT-KAGYAは、推論・loss計算・memory・sleep-time consolidation・QLoRA・adapter evaluationまでを、KAGYA自身のコードだけで完結する独立した実行環境として運用したい。

加えて、KAGYAの高速な内部判断をすべてGemma 4 E4Bへ委譲するのではなく、0.6B級の小型decision modelをSystem Oneとして同居させたい。

最終的なhardware contractは以下を目標とする。

- NVIDIA GPU 1枚
- VRAM 24 GiB
- KAGYA本体とSystem OneがAWAKE時に同時常駐可能
- SLEEP時には推論系をunloadし、同じGPUでQLoRAを実行可能
- homelab-llm-platform等の外部worker runtimeをKAGYAノード内へ持ち込まない
- PROJECT-KAGYA repositoryだけでKAGYAノードを構築・再現可能にする

## 目的

KAGYA専用のSystem One decision modelをPROJECT-KAGYA内部に実装し、Gemma 4 E4BをSystem Twoとして補完する。

初期backbone候補としてQwen3-0.6B級を想定する。NanoJev / Kev等の実装はreferenceとして利用できるが、最終的にはKAGYA固有stateとdecision taskに特化したモデルとする。

## 想定アーキテクチャ

```text
RTX 3090 24GB
│
├─ AWAKE
│   ├─ System Two
│   │   └─ Gemma 4 E4B NF4 + ACTIVE LoRA
│   │
│   └─ System One
│       └─ KAGYA-Jev 0.6B
│           ├─ boolean
│           ├─ choice
│           └─ ordered score
│
└─ SLEEP
    ├─ unload System Two inference
    ├─ unload System One
    ├─ dream dataset generation
    ├─ Gemma 4 E4B QLoRA
    ├─ candidate adapter evaluation
    ├─ optional System One training
    └─ reload AWAKE runtime
```

System Oneは別HTTP serviceとしてではなく、PROJECT-KAGYA内部のPython componentとして直接呼び出す。

例:

```python
decision = system_one.decide(
    state=state,
    questions=[
        BooleanQuestion(...),
        ChoiceQuestion(...),
        ScoreQuestion(...),
    ],
)
```

## 初期decision scope

v1ではSystem Oneの責務を高速で低リスクな判断に限定する。

候補:

- `needs_deep_reasoning: bool`
- `memory_retrieval_depth: score(0..3)`
- `interaction_mode: choice`
- `experience_importance: score(0..4)`
- `needs_user_clarification: bool`
- `tool_intent: choice`
- `response_urgency: score`
- `novelty: score`

以下はSystem One単独では決定しない。

- Value state mutation
- canonical memory deletion
- adapter activation
- tool execution
- other irreversible / authority-bearing state transitions

System Oneの出力はproposal / evidence / probabilityとして扱い、最終authorityはKAGYA Runtime側に残す。

## System One / System Two fallback

System Oneはfail-openではなく、System Twoへfallbackできる構造とする。

```text
Jev confident
  -> fast decision

Jev uncertain
  -> System Two

Jev unavailable
  -> System Two

Jev invalid
  -> System Two
```

confidence thresholdは固定の経験則ではなく、KAGYA固有validation setでcalibrationした値を使用する。

## 学習

System TwoとSystem Oneの学習objectiveは分離する。

### System Two

- Gemma 4 E4B
- SFT / QLoRA
- dream dataset
- behavior / reasoning / memory-use adaptation

### System One

- 0.6B級backbone + decision heads
- boolean / choice / ordered-score classification
- probability calibration
- KAGYA固有decision trajectoryから学習

想定dataset source:

- KAGYA internal state
- System One proposal
- actual System Two decision
- human feedback
- later correction
- observed outcome

概念:

```text
KAGYA experience
    ↓
System Two / actual outcome / feedback
    ↓
decision dataset
    ↓
KAGYA-Jev training
    ↓
improved System One
```

Gemma側LoRAとSystem One側LoRA/decision headは別artifact・別lifecycleとして扱う。

## Process isolation

同一Python process内で、

```python
del model
torch.cuda.empty_cache()
```

だけに依存してinference/trainingを切り替えない。

PROJECT-KAGYA内部でsubprocess境界を利用し、process terminationをVRAM cleanup boundaryとして使える構造を検討する。

例:

```text
kagya-supervisor
├─ inference process
├─ system-one process/component
├─ trainer process
└─ evaluator process
```

ただし、すべてPROJECT-KAGYAのコードとして実装する。

## 24GB Hardware Acceptance Contract

RTX 3090 24GB級1枚で以下を満たすことを目標とする。

### AWAKE

- [ ] Gemma 4 E4BをNF4でロードできる
- [ ] ACTIVE LoRAをロードできる
- [ ] KAGYA-Jev 0.6Bを同時常駐できる
- [ ] System One multi-question decisionを実行できる
- [ ] 通常のSystem Two推論を実行できる
- [ ] peak VRAMを23 GiB未満に抑える

### SLEEP

- [ ] inference modelを安全にunloadできる
- [ ] System Oneをunloadできる
- [ ] 同一24GB GPU上でGemma 4 E4B QLoRAを実行できる
- [ ] candidate adapterを生成できる
- [ ] candidate evaluationを実行できる
- [ ] 必要に応じてSystem One trainingを実行できる
- [ ] AWAKE runtimeへ復帰できる

## QLoRA初期profile

24GB single-GPUを優先し、初期値は保守的にする。

- NF4
- double quantization
- BF16 compute
- LoRA rank 16
- batch size 1
- gradient accumulation
- gradient checkpointing
- `use_cache = false`
- 初期max sequence lengthは2048程度
- multimodal QLoRAは初期scope外

sequence length等は実測VRAMを見ながら段階的に拡張する。

## 非目標

- OpenJev / DiffusionGemmaをKAGYAの常駐System Oneとして採用すること
- homelab-llm-platform worker runtimeをKAGYAノードへ導入すること
- System Oneへcanonical state mutation authorityを与えること
- 初期段階でmultimodal System One trainingを行うこと
- 24GBを超えることを前提としてmulti-GPU化すること

## 完了条件

- [ ] System One API / schema / decision-head contractが定義されている
- [ ] 0.6B級reference implementationがPROJECT-KAGYA内で動作する
- [ ] System Two fallbackが実装されている
- [ ] System Oneのdecision dataset formatが定義されている
- [ ] System One training/evaluation pathが定義されている
- [ ] System Oneにauthorityを与えないruntime gateが存在する
- [ ] 24GB hardware acceptance testが文書化されている
- [ ] AWAKE -> SLEEP -> AWAKEのVRAM lifecycleが再現可能である
- [ ] KAGYAノードがPROJECT-KAGYAのみで構築可能である
